### ISSUE TITLE
Add support for NPCs

### ISSUE THREAD

**Issue by jgamedev** [arjanfrans/realtime-multiplayer-in-html5#7]
NPCs are super fun!! :laughing: :laughing: 

Sorry, I know this doesn't help much.
I will update this issue with more useful comments once I understand the codebase better but I wanted to submit this so I can come back to it later.

**Comment by jptosso** [comment]
For NPCs I would just use the player class and create a bot, instead of
receiving inputs by a browser, receive inputs by the server :)

2016-07-03 11:30 GMT-04:00 jgamedev notifications@github.com:

**Comment by jgamedev** [comment]
@asdfuken Do you have any examples of this approach being used anywhere?

Well, regular socket communication between both server would definitely add up to the total data exchange payload, unless you use a separate node process for the bot server and communicate via IPC with the game server or some similar type of solution.

However, since bot AI is expensive, offloading it from the game server would definitely be a huge plus.

**Comment by jptosso** [comment]
I agree with you, I would add a second server and IPC communication as it
would rather slow down the server main thread.
It could be written in the same server code as a Asynchronous activity or a
new server.
For communication between processes I would register each NPC with a
special UUID as a common player (Browser Like), so you wouldn't have to
make hard modifications for the client.

2016-07-03 15:01 GMT-04:00 jgamedev notifications@github.com:
